### PR TITLE
Filter friendly name

### PR DESCRIPTION
**Pull request by wanjunsli**

### CODE REVIEW

File: src/test/java/com/twilio/sdk/resource/instance/taskrouter/WorkflowTest.java [hunk 1]
```diff
@@ -95,6 +95,29 @@ public void testCreateWorkflowWithHelper() throws Exception {
...
+                WorkflowConfiguration workflowConfiguration = workflow.parseConfiguration();
+                List<WorkflowRule> workflowRules = workflowConfiguration.getWorkflowRules();
+
+                assertEquals(workflowRules.get(0).getFriendlyName(), "Sales");
+                assertEquals(workflowRules.get(0).getFilterFriendlyName(), "Sales");
+                assertEquals(workflowRules.get(1).getFriendlyName(), "Marketing");
+                assertEquals(workflowRules.get(1).getFilterFriendlyName(), "Marketing");
+                assertEquals(workflowRules.get(2).getFriendlyName(), "Support");
+                assertEquals(workflowRules.get(2).getFilterFriendlyName(), "Support");
+        }
```
Comment (Ragil) [inline src/test/java/com/twilio/sdk/resource/instance/taskrouter/WorkflowTest.java]: spaces -> tabs

File: src/test/java/com/twilio/sdk/resource/instance/taskrouter/WorkflowTest.java [hunk 2]
```diff
@@ -95,6 +95,29 @@ public void testCreateWorkflowWithHelper() throws Exception {
...
+                Workflow workflow = taskRouterClient.createWorkflow("WSaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", properties);
+                WorkflowConfiguration workflowConfiguration = workflow.parseConfiguration();
+                List<WorkflowRule> workflowRules = workflowConfiguration.getWorkflowRules();
+
+                assertEquals(workflowRules.get(0).getFriendlyName(), "Sales");
+                assertEquals(workflowRules.get(0).getFilterFriendlyName(), "Sales");
+                assertEquals(workflowRules.get(1).getFriendlyName(), "Marketing");
+                assertEquals(workflowRules.get(1).getFilterFriendlyName(), "Marketing");
+                assertEquals(workflowRules.get(2).getFriendlyName(), "Support");
+                assertEquals(workflowRules.get(2).getFilterFriendlyName(), "Support");
```
Comment (Ragil) [inline src/test/java/com/twilio/sdk/resource/instance/taskrouter/WorkflowTest.java]: we should also test that the same workflow is marshaled back as friendly_name instead of filter_friendly_name

File: src/test/java/com/twilio/sdk/resource/instance/taskrouter/WorkflowTest.java [hunk 3]
```diff
@@ -95,6 +95,45 @@ public void testCreateWorkflowWithHelper() throws Exception {
...
+		Workflow workflow2 = taskRouterClient.createWorkflow("WSaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", properties2);
+		WorkflowConfiguration workflowConfiguration2 = workflow2.parseConfiguration();
+
+		assertEquals(workflowConfiguration2.toJSON(), configurationStrWithFriendlyName);
+	}
+	
+	@Test
+	public void testWorkflowWithFilterFriendlyName() throws Exception {
+		setExpectedServerReturnCode(201);
+		setExpectedServerAnswer("/" + getClass().getPackage().getName().replace(".", "/") + "/workflow_filter_friendly_name.json");
```
Comment (Ragil) [inline src/test/java/com/twilio/sdk/resource/instance/taskrouter/WorkflowTest.java]: are these two lines related to the test?

File: src/test/java/com/twilio/sdk/resource/instance/taskrouter/WorkflowTest.java [hunk 4]
```diff
@@ -95,6 +95,45 @@ public void testCreateWorkflowWithHelper() throws Exception {
 		assertNull(workflow.getFallbackAssignmentCallbackUrl());
 		assertEquals(120, workflow.getTaskReservationTimeout().intValue());
 	}
+	
+	@Test
+		public void testSerializeWorkflowFriendlyName() throws Exception {
+		setExpectedServerReturnCode(201);
```
Comment (Ragil) [inline src/test/java/com/twilio/sdk/resource/instance/taskrouter/WorkflowTest.java]: is this related to the test?

File: src/test/java/com/twilio/sdk/resource/instance/taskrouter/WorkflowTest.java [hunk 5]
```diff
@@ -95,6 +95,45 @@ public void testCreateWorkflowWithHelper() throws Exception {
 		assertNull(workflow.getFallbackAssignmentCallbackUrl());
 		assertEquals(120, workflow.getTaskReservationTimeout().intValue());
 	}
+	
+	@Test
+		public void testSerializeWorkflowFriendlyName() throws Exception {
```
Comment (Ragil) [inline src/test/java/com/twilio/sdk/resource/instance/taskrouter/WorkflowTest.java]: this test validates that `friendly_name` is re-serialized into `friendly_name` but we should really be testing that `filter_friendly_name` into `friendly_name` - the intended fix to the bug

Comment (wanjunsli) [inline src/test/java/com/twilio/sdk/resource/instance/taskrouter/WorkflowTest.java]: I split the test I initially had into two -- one on friendly_name and one on filter_friendly_name. I made some changes though to make it more readable.

**Comment by Ragil** [comment]
+1
